src/store/actions/Chat.ts

```diff
@@ -1,161 +1,52 @@
-  deleted?: boolean;
-};
-
-type NonGlobalChatMessage = BaseChatMessage & {
-  to: string;
-};
-
-export type GlobalChatMessage = BaseChatMessage & {
-  type: ChatMessageType.Global;
-};
-
-export type RestrictedChatMessage = NonGlobalChatMessage & {
-  type: ChatMessageType.Table | ChatMessageType.Room;
-};
-
-export type PrivateChatMessage = NonGlobalChatMessage & {
-  type: ChatMessageType.Private;
-  isRead: boolean;
-};
-
-export type ChatMessage =
-  | GlobalChatMessage
-  | PrivateChatMessage
-  | RestrictedChatMessage;
-
-const collectionName = (venueId: string) => `venues/${venueId}/chats`;
-
-const buildMessage = (
-  data: BuildMessageInput | BuildMessageToInput
-): ChatMessage => {
-  const base: BaseChatMessage = {
-    ts_utc: firebase.firestore.Timestamp.fromDate(new Date()),
-    text: data.text,
-    from: data.from,
-  };
-
-  switch (data.type) {
-    case ChatMessageType.Global:
-      return { ...base, type: data.type };
-    case ChatMessageType.Room:
-    case ChatMessageType.Table:
-      return { ...base, type: data.type, to: data.to };
-    case ChatMessageType.Private:
-      return { ...base, type: data.type, to: data.to, isRead: false };
-  }
-};
-
-const saveMessage = async (venueId: string, message: ChatMessage) =>
-  await firebase
-    .firestore()
-    .collection(collectionName(venueId))
-    .add(buildMessage(message));
-
-export const sendGlobalChat = createAsyncThunk<void, SendGlobalChatFields>(
-  SEND_GLOBAL_CHAT,
-  async ({ venueId, text, from }) => {
-    await saveMessage(
-      venueId,
-      buildMessage({ type: ChatMessageType.Global, text, from })
-    );
-  }
-);
-
-export const sendPrivateChat = createAsyncThunk<void, SendPrivateChatFields>(
-  SEND_PRIVATE_CHAT,
-  async ({ text, from, to }) => {
-    for (const messageUser of [from, to]) {
-      await firebase
-        .firestore()
-        .collection("privatechats")
-        .doc(messageUser)
-        .collection("chats")
-        .add(buildMessage({ type: ChatMessageType.Private, text, from, to }));
-    }
-  }
-);
-
-export const sendRoomChat = createAsyncThunk<void, SendRoomChatFields>(
-  SEND_ROOM_CHAT,
-  async ({ venueId, text, from, to }) => {
-    await saveMessage(
-      venueId,
-      buildMessage({ type: ChatMessageType.Room, text, from, to })
-    );
-  }
-);
 
-export const sendTableChat = createAsyncThunk<void, SendTableChatFields>(
-  SEND_TABLE_CHAT,
-  async ({ venueId, text, from, to }) => {
-    await saveMessage(
-      venueId,
-      buildMessage({ type: ChatMessageType.Table, text, from, to })
-    );
-  }
-);
+export type SetChatsSidebarVisibilityAction = ReduxAction<
+  ChatActionTypes.SET_CHAT_SIDEBAR_VISIBILITY,
+  { isVisible: boolean }
+>;
+
+export type SetVenueChatTabOpenedAction = ReduxAction<
+  ChatActionTypes.SET_VENUE_CHAT_TAB_OPENED,
+  VenueChatSettings
+>;
+
+export type SetPrivateChatTabOpenedAction = ReduxAction<
+  ChatActionTypes.SET_PRIVATE_CHAT_TAB_OPENED,
+  PrivateChatSettings
+>;
```

Comment:
Redux actions shouldn't be written as 'setter's:

- https://redux.js.org/style-guide/style-guide#model-actions-as-events-not-setters